map/local_ads_helpers.cpp

```diff
@@ -0,0 +1,92 @@
+#include "map/local_ads_helpers.hpp"
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "base/assert.hpp"
+#include "base/bits.hpp"
+
+namespace
+{
+template<typename TPrimitive>
+TPrimitive ReadPrimitive(FileReader const & reader, uint64_t & offset)
```

Comment:
Why not to use ReaderSource instead of these manipulations with raw reader and offset? The code will be simpler, our API for reading/writing primitives works nicely with ReaderSources.